docs/tiltfile_concepts.md

```diff
@@ -144,38 +144,95 @@ can be expanded and investigated.
+resources. For more information, see the [Docker Compose documentation](docker_compose.html).)
+
+### Configuring Kubernetes Resources
 In many cases, Tilt's automatic resource assembly logic will be sufficient for you to run
 your app. However, if you need to configure your Kubernetes resources on top of Tilt's
 automatic assembly, you can do so with a call to [`k8s_resource`](api.html#api.k8s_resource).
-Here we'll discuss the most common configuration arguments: `new_name` and `port_forwards`.
-For discussion of other available arguments, see the [api spec](api.html#api.k8s_resource).
 
-`new_name` allows you to specify a new resource name, in case you do not like the automatically generated one.
+Here we'll discuss the most common configuration options for Kubernetes resources:
+`new_name`, `port_forwards`, and `objects`. For discussion of other available arguments, see
+the [API spec](api.html#api.k8s_resource).
+
+`new_name` allows you to specify a new resource name, in case you do not like the
+automatically generated one:
 
 ```python
 # rename the resource "redis:deployment" to "redis"
-k8s_resource('redis:deployment', new_name='redis')
+k8s_resource(workload='redis:deployment', new_name='redis')
 ```
 
+(Use this pattern to rename individual resources; to programmatically rename all resources,
+see [`workload_to_resource_function`](api.html#api.k8s_resource).)
+
 Tilt also supports a few ways to specify `port_forwards`:
 
 ```python
 # connect localhost:9000 to the default container port
-k8s_resource('frontend', port_forwards=9000)
+k8s_resource(
+  workload='frontend',
+  port_forwards=9000
+)
 
 # connect localhost:9000 to container port 8000
-k8s_resource('frontend', port_forwards='9000:8000')
+k8s_resource(
+  workload='frontend',
+  port_forwards='9000:8000'
+)
 
 # connect localhost:9000 to container port 8000
 # and localhost:9001 to container port 8001
-k8s_resource('frontend', port_forwards=['9000:8000', '9001:8001'])
+k8s_resource(
+  workload='frontend',
+  port_forwards=['9000:8000', '9001:8001']
+)
+```
+
+Additionally, you may want to add additional Kubernetes objects to an existing
+resource (say, group a Secret with the Deployment that makes use of it), or group
+non-workload objects into their own resource (e.g. make a CRD its own resource X,
+so a workload containing an instance of that CR may name X as a dependency). To
+accomplish this, use the `k8s_resource.objects` parameter to specify one or more
+Kubernetes objects.
+
+```python
+# associate an existing Secret and Volume with the "frontend" service
+k8s_resource(
+  workload='frontend',
+  objects=['frontend:secret', 'frontend:volume']
+)
+
+# make a new resource consisting of some objects necessary for cluster setup
+k8s_resource(
+  objects=['my-ns:namespace', 'kafka:crd', 'some-ingress:ingress'],
+  new_name='cluster-setup',
+)
 ```
 
+#### Kubernetes Object Selectors
+
+Tilt specifies Kubernetes objects via _Kubernetes object selectors_.
```

Comment:
I think selector already means something in the Kubernetes ecosystem, so I'd suggest using a different term.

Reply:
Hmm okay I rethought this term a bit, and it DOES feel like "selector" is the most accurate word than other word I (or @landism ) could think of (e.g. "identifier", "matcher", "query"). So I'd propose (as I think DB mentioned in our 1:1) making a _phrase_ that we always use for this type so it's not just "selector":
* "object selector"
* gvk selector"
* some proper name like "Tilt Kubernetes Object Selector" (TKOS)

anyone have particular thoughts here? @dbentley @jazzdan @landism 

bonus: my sweetie suggested "identifier chain" (b/c you have bits of identifier that you chain together in a specific order--and you can have a full identifier chain or a partial one). It sounds sounds buzzwordy and maybe a little too complex but I find myself strangely taken with it anyway. If anyone else is ALSO strangely taken with it, then it might be worth considering. if no one else is, then ignore me 😅

Reply:
I like identifier chain, though as @dbentley points out technically it isn't dealing with "identity" so much as "specification". Specifier chain? It IS a dope phrase.

I like object selector of the more boring ones you outlined above, and IMO the fact that those feel boring to me means that we should go with those.

Reply:
Ha okay we'll keep "specifier chain" on the backburner if we get negative feedback for "object selector" :)